server/handlers.go

```diff
@@ -583,6 +583,35 @@ func (s *Server) Unlock(token, filename string) error {
 	return nil
 }
 
+func (s *Server) GetMetadata(token, filename string) (Metadata, error) {
```

Comment:
please, don't add a 99% identical method (see `CheckMetadata`), just refactor the existing one to return `Metadata`

also related to head requests increasing the download count `CheckMetadata` can become:
`func (s *Server) ProcessMetadata(token, filename string, increaseDownload bool) (Metadata, error) {`